packages/react/src/BlockNoteView.tsx

```diff
@@ -1,33 +1,97 @@
-import { BlockNoteEditor, BlockSchema } from "@blocknote/core";
-import { MantineProvider } from "@mantine/core";
+import { BlockNoteEditor, BlockSchema, mergeCSSClasses } from "@blocknote/core";
+import { createStyles, MantineProvider } from "@mantine/core";
 import { EditorContent } from "@tiptap/react";
-import { HTMLAttributes, ReactNode } from "react";
-import { getBlockNoteTheme } from "./BlockNoteTheme";
+import { HTMLAttributes, ReactNode, useEffect, useMemo, useState } from "react";
+import { blockNoteToMantineTheme, Theme } from "./BlockNoteTheme";
 import { FormattingToolbarPositioner } from "./FormattingToolbar/components/FormattingToolbarPositioner";
 import { HyperlinkToolbarPositioner } from "./HyperlinkToolbar/components/HyperlinkToolbarPositioner";
 import { SideMenuPositioner } from "./SideMenu/components/SideMenuPositioner";
 import { SlashMenuPositioner } from "./SlashMenu/components/SlashMenuPositioner";
+import { darkDefaultTheme, lightDefaultTheme } from "./defaultThemes";
+
+// Renders the editor as well as all menus & toolbars using default styles.
+function BaseBlockNoteView<BSchema extends BlockSchema>(
+  props: {
+    editor: BlockNoteEditor<BSchema>;
+    children?: ReactNode;
+  } & HTMLAttributes<HTMLDivElement>
+) {
+  const { classes } = createStyles({ root: {} })(undefined, {
+    name: "Editor",
+  });
+
+  const { editor, children, className, ...rest } = props;
+
+  return (
+    <EditorContent
+      editor={props.editor?._tiptapEditor}
+      className={mergeCSSClasses(classes.root, props.className || "")}
+      {...rest}>
+      {props.children || (
+        <>
+          <FormattingToolbarPositioner editor={props.editor} />
+          <HyperlinkToolbarPositioner editor={props.editor} />
+          <SlashMenuPositioner editor={props.editor} />
+          <SideMenuPositioner editor={props.editor} />
+        </>
+      )}
+    </EditorContent>
+  );
+}
 
 export function BlockNoteView<BSchema extends BlockSchema>(
   props: {
     editor: BlockNoteEditor<BSchema>;
+    theme?:
+      | "light"
+      | "dark"
+      | Theme
+      | {
+          light: Theme;
+          dark: Theme;
+        };
     children?: ReactNode;
   } & HTMLAttributes<HTMLDivElement>
 ) {
-  const { editor, children, ...rest } = props;
+  const {
+    theme = { light: lightDefaultTheme, dark: darkDefaultTheme },
+    ...rest
+  } = props;
+
+  const [browserTheme, setBrowserTheme] = useState<"light" | "dark">(
```

Comment:
maybe use a separate hook like https://github.com/rfoel/use-prefers-color-scheme/blob/main/src/index.ts to isolate this logic?